### DISCUSSION TITLE
How do I bind stuff to the option key on MacOS?

### DISCUSSION THREAD

**Discussion by Hubro** [neovide/neovide#1270] [Q&A]
I assume this is possible, since I see a lot of people using Neovide on MacOS, but I am completely unable to bind stuff to the option key on MacOS. Instead of being treated as the meta key, it just produces symbols.

iTerm and kitty have a setting to treat the left option key as Alt, which makes keybinds portable from other operating systems.

Does Neovide have a setting like this that I missed?

**Comment by rikiwarren** [comment]
I'm having the same problem. I've tried the following. None of these work:

```

" These work in nvim but not Neovide.
tnoremap <M-h> <C-\><C-n>gT
tnoremap <M-l> <C-\><C-n>gt
noremap <M-h> gT
noremap <M-l> gt

" I tried these on Neovide, none worked.
tnoremap <^[h> <C-\><C-n>gT
tnoremap <^[l> <C-\><C-n>gt
noremap <^[h> gT
noremap <^[l> gt

tnoremap <˙> <C-\><C-n>gT
tnoremap <¬> <C-\><C-n>gt
noremap <˙> gT
noremap <¬> gt
```

**Comment by secheng722** [comment]
setting let g:neovide_input_macos_alt_is_meta = v:true fixed the problem

**Comment by apettel** [comment]
I'm using the [EurKEY Keyboard Layout](https://eurkey.steffen.bruentjen.eu/?lang=en). To access certain characters (e.g. ä, ö, ü, ß) I need the alt key. But I also need the alt key as meta key for certain terminal applications. 
I configured the right alt key in my terminal program to switch the characters in the EurKEY-Layout and the left alt key is used as meta key. I created a patch for neovide to allow the separate configuration of the right and the left alt key (branch: macos-left-right-alt-meta).

[GitHub apettel neovide fork](https://github.com/apettel/neovide/tree/macos-left-right-alt-meta)

**Comment by 9mm** [comment]
I use the option key.

```lua
  vim.g.neovide_input_macos_alt_is_meta = true
```

Here i use it in barbar to move tabs:

```lua
    vim.keymap.set({ 'n' }, '<M-[>', function() vim.cmd('BufferMovePrevious') end, { desc = 'Move tab left' })
    vim.keymap.set({ 'n' }, '<M-]>', function() vim.cmd('BufferMoveNext') end, { desc = 'Move tab right' })
```

**Reply by duduomena1** [reply]
I use 
<code> vim.g.neovide_input_macos_option_key_is_meta = 'only_left'</code>
That work's for me

**Comment by sanmiguel** [comment]
I use `D` mappings instead of `M` mappings for this, works just fine. e.g. `vnoremap <D-X> "+d`
